### PR TITLE
feat(unity): Automated symbol upload

### PR DESCRIPTION
**Pull request by bitsandfoxes**
The Unity SDK now provides automated symbol upload for both Android and iOS.

### CODE REVIEW

File: src/platforms/unity/native-support.mdx [hunk 1]
```diff
@@ -18,19 +18,21 @@ You can opt out of native support by unchecking `iOS Native Support` and `Androi
 
 ## Debug Symbols
 
-Sentry requires [debug information files](/platforms/android/data-management/debug-files/) to symbolicate your crash logs. To upload these, the Unity SDK relies on the [sentry-cli](/product/cli/). We included the executables for Windows, macOS, and Linux inside the Sentry SDK package.
-
-### Android
+Sentry requires [debug information files](/platforms/android/data-management/debug-files/) to symbolicate your crash logs. The Unity SDK provides an automated upload functionality for those symbol files that relies on the [sentry-cli](/product/cli/). We included the executables for Windows, macOS, and Linux inside the Unity SDK package.
```
Comment (imatwawana) [inline src/platforms/unity/native-support.mdx:21]:
```suggestion
Sentry requires [debug information files](/platforms/android/data-management/debug-files/) to symbolicate your crash logs. The Unity SDK provides an automated upload functionality for those symbol files that rely on the [sentry-cli](/product/cli/). We've included the executables for Windows, macOS, and Linux inside the Unity SDK package.
```

File: src/platforms/unity/native-support.mdx [hunk 2]
```diff
@@ -18,19 +18,21 @@ You can opt out of native support by unchecking `iOS Native Support` and `Androi
...
 
 The automated debug symbols upload is enabled by default but requires configuration. Go to **Tools > Sentry > Editor** to enter the [Auth Token](https://sentry.io/api/), Org Slug, and the Project Name. Note that the Unity SDK creates a file at `Assets/Plugins/Sentry/SentryCliOptions.asset` to store the configuration, that should not be made publicly available.
 
-### iOS
+### iOS - dSYM and Bitcode
 
 Debug information files on the iOS platform are called dSYM, and the way to obtain them differs depending on whether `Enable Bitcode` is set to `true` in your Xcode project.
 
-With Bitcode disabled, make sure the `Debug Information Format` is set to `DWARF with dSYM file`. When building the game, Xcode will then place the dSYM folders together with the build output and you can run the sentry-cli command there. By default the output path will look similar to: _~/Library/Developer/Xcode/DerivedData/BUILD_ID/Build/Products/BUILD_TYPE/_.
+For Sentry to symbolicate your crash logs and with `bitcode` enabled we need two types of files:
```
Comment (imatwawana) [inline src/platforms/unity/native-support.mdx:29]:
```suggestion
For Sentry to symbolicate your crash logs and with `bitcode` enabled, we need two types of files:
```

File: src/platforms/unity/native-support.mdx [hunk 3]
```diff
@@ -18,19 +18,21 @@ You can opt out of native support by unchecking `iOS Native Support` and `Androi
...
+1. `dSYM` files available only **after** App Store Connect finishes processing the build
+2. `BCSymbolMap` files that get created during the archiving process
```
Comment (imatwawana) [inline src/platforms/unity/native-support.mdx:31]:
```suggestion
2. `BCSymbolMap` files that are created during the archiving process
```

File: src/platforms/unity/native-support.mdx [hunk 4]
```diff
@@ -18,19 +18,21 @@ You can opt out of native support by unchecking `iOS Native Support` and `Androi
...
+
+The automated symbol upload will take care of the `BCSymbolMap` files by processing them during the archiving process. To provide the dSYM files to Sentry you can either set up the [App Store Connect Integration](/platforms/apple/guides/ios/dsym/#bitcode-appstore) so Sentry can fetch them for you or download and then upload them [manually using sentry-cli](/platforms/apple/guides/ios/dsym/#bitcode-sentrycli).
```
Comment (imatwawana) [inline src/platforms/unity/native-support.mdx:33]:
```suggestion
The automated symbol upload will take care of the `BCSymbolMap` files by processing them during the archiving process. To provide the dSYM files to Sentry, you can either set up the [App Store Connect Integration](/platforms/apple/guides/ios/dsym/#bitcode-appstore) so Sentry can fetch them for you or download and then upload them [manually using sentry-cli](/platforms/apple/guides/ios/dsym/#bitcode-sentrycli).
```

File: src/platforms/unity/native-support.mdx [hunk 5]
```diff
@@ -18,19 +18,21 @@ You can opt out of native support by unchecking `iOS Native Support` and `Androi
...
 
-If you have Bitcode enabled, please refer to our [iOS documentation](/platforms/apple/guides/ios/dsym/).
+With `bitcode` disabled the automated symbols upload will pick up the `dSYM` files at the end of the build process without further action required.
```
Comment (imatwawana) [inline src/platforms/unity/native-support.mdx:35]:
```suggestion
With `bitcode` disabled, the automated symbols upload will pick up the `dSYM` files at the end of the build process without further action required.
```

**Review by imatwawana — APPROVED** [review]
Minor fixes, otherwise looks good!

**Comment by bitsandfoxes** [comment]
> Minor fixes, otherwise looks good!

Thanks a lot for all the time and effort! (and patience with the commas. I promise I'll learn eventually :D)

**Comment by imatwawana** [comment]
> > Minor fixes, otherwise looks good!
> 
> Thanks a lot for all the time and effort! (and patience with the commas. I promise I'll learn eventually :D)

Ha! No worries, commas are hard!

**Comment by bruno-garcia** [comment]
0.8.0 is just out: https://github.com/getsentry/unity

**Pull request merged**